File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.go
func evalRPN(tokens []string) int {
	var stack []string

	const (
		plus  = "+"
		minus = "-"
		mul   = "*"
		div   = "/"
	)

	for _, token := range tokens {
		switch token {
		case plus:
			a, _ := strconv.Atoi(stack[len(stack)-2])
			b, _ := strconv.Atoi(stack[len(stack)-1])
			stack = stack[:len(stack)-2]
			stack = append(stack, strconv.Itoa(a+b))
		case minus:
			a, _ := strconv.Atoi(stack[len(stack)-2])
			b, _ := strconv.Atoi(stack[len(stack)-1])
			stack = stack[:len(stack)-2]
			stack = append(stack, strconv.Itoa(a-b))
		case mul:
			a, _ := strconv.Atoi(stack[len(stack)-2])
			b, _ := strconv.Atoi(stack[len(stack)-1])
			stack = stack[:len(stack)-2]
			stack = append(stack, strconv.Itoa(a*b))
		case div:
			a, _ := strconv.Atoi(stack[len(stack)-2])
			b, _ := strconv.Atoi(stack[len(stack)-1])
			stack = stack[:len(stack)-2]
			stack = append(stack, strconv.Itoa(a/b))
		default:
			stack = append(stack, token)
		}
	}
	result, _ := strconv.Atoi(stack[0])
	return result
}

File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.rs
impl Solution {
    pub fn eval_rpn(tokens: Vec<String>) -> i32 {
        let mut stack: Vec<String> = Vec::new();

        for token in tokens {
            match token.as_str() {
                "+" => {
                    let a = stack.pop().unwrap().parse::<i32>().unwrap();
                    let b = stack.pop().unwrap().parse::<i32>().unwrap();
                    stack.push((b+a).to_string())
                }
                "-" => {
                    let a = stack.pop().unwrap().parse::<i32>().unwrap();
                    let b = stack.pop().unwrap().parse::<i32>().unwrap();
                    stack.push((b-a).to_string())
                }
                "*" => {
                    let a = stack.pop().unwrap().parse::<i32>().unwrap();
                    let b = stack.pop().unwrap().parse::<i32>().unwrap();
                    stack.push((b*a).to_string())
                }
                "/" => {
                    let a = stack.pop().unwrap().parse::<i32>().unwrap();
                    let b = stack.pop().unwrap().parse::<i32>().unwrap();
                    stack.push((b/a).to_string())
                }
                _ => stack.push(token)
            }
        }

        stack[0].parse::<i32>().unwrap()
    }
}